doc/nrf/app_dev/data_storage.rst

```diff
@@ -0,0 +1,498 @@
+   :alt: Data storage using NVS
+
+   Data storage using NVS
+
+This option offers the following features:
+
+* Simple API - Basic functionality requires only three functions: ``nvs_mount()``, ``nvs_write()``, and ``nvs_read()``.
+* Automatic integrity - Uses CRC to verify data integrity automatically.
+* Wear leveling - Distributes writes across flash sectors to extend memory lifetime.
+* Atomic operations - Ensures data consistency during power loss.
+
+NVS automatically manages metadata for stored elements, including the following components:
+
+* Element ID
+* Data offset in the sector
+* Data length
+* CRC for integrity checking
+
+During initialization, NVS verifies all stored data and ignores any data with missing or incorrect metadata.
+
+Configuration
+-------------
+
+To enable NVS, set the following Kconfig options:
+
+* :kconfig:option:`CONFIG_NVS`
+* :kconfig:option:`CONFIG_FLASH`
+* :kconfig:option:`CONFIG_FLASH_MAP`
+
+Usage example
+-------------
+
+See Zephyr's :zephyr:code-sample:`nvs` sample for an example of how to use NVS.
+
+.. _data_storage_rramc:
+
+RRAMC (Resistive Random Access Memory Controller)
+=================================================
+
+.. note::
+   RRAMC is only available on the `nRF54L15 SoC <nRF54L15 RRAMC_>`_ and should be used with :ref:`ZMS <data_storage_zms>`.
+
+.. list-table:: RRAMC features
+   :widths: auto
+   :header-rows: 1
+
+   * - Feature
+     - Partitioning
+     - Integrity
+     - Isolation
+     - Encryption
+   * - **RRAMC**
+     - Yes
+     - Yes
+     - Yes
+     - No
+
+The Resistive Random Access Memory Controller (RRAMC) provides advanced functionality for managing RRAM memory, including the secure information configuration region (SICR) and user information configuration registers (UICR).
+
+.. figure:: /images/data_storage_rramc.svg
+   :alt: Data storage using RRAMC
+
+   Data storage using RRAMC
+
+This option offers the following features:
+
+* Write without erase - RRAMC allows writing and overwriting without requiring erase operations.
+* Error correction - Built-in ECC (Error Correction Code) can detect and correct up to two bit errors per 128-bit word line.
+* Power management - Automatic standby or power-down modes with configurable latency.
+* Region protection - Configurable memory regions with read, write, execute, and secure access controls.
+* Write-once support - Optional write-once protection for memory regions.
+* Buffered writes - Support for fast buffered writes to contiguous memory regions.
+* Power-failure protection - Built-in power-fail comparator to prevent writes during low voltage conditions.
+
+The RRAMC provides several protection mechanisms, including:
+
+* Region protection - Memory can be divided into configurable regions with different access permissions.
+* Immutable boot region - Configurable read-only boot region starting at address ``0x00000000``.
+* One-time programmable protection - Optional write-once protection for UICR.
```

Comment:
```suggestion
* One-time programmable protection - Optional write-once protection for UICR.
```
```suggestion
* One-time programmable protection (Hardware-fixed for UICR).
```

RRAMC regions can be configured as write-once (optionally), but UICR is hardware-fixed to be write-once